### ISSUE TITLE
Moving files to trash always fails in Kubuntu 16.04

### ISSUE THREAD

**Issue by kbumsik** [Microsoft/vscode#22820]
Deleting files always fails in Kubuntu 16.04.
I wish if there is any option to delete files permanently at the beginning.

- VSCode Version: Version 1.10.2
- OS Version: Kubuntu 16.04.2 LTS

Steps to Reproduce:

1. Right click to a file on the left Explorer tab
2. Click `Delete`
3. Prompt to confirm appears and then click `move to trash`.
4. Error message `failed to move to trash` appears

**Comment by bpasero** [comment]
@kbumsik you should be able to Shift+Delete to bypass the trash

@Tyriar any ideas? does the user maybe have to install some library to get support for the trash?

**Comment by Tyriar** [comment]
Hmm, I haven't heard anything like this before. @kbumsik are you running in a corporate network that might have special permissions on certain directories?

@bpasero all the trash logic is contained within node's `fs` module isn't it?

**Comment by bpasero** [comment]
@Tyriar no, the trash logic is coming in from Electron and there seem to be bugs, e.g. https://github.com/electron/electron/issues/8132, https://github.com/electron/electron/issues/8124 and https://github.com/electron/electron/issues/4974

**Comment by Tyriar** [comment]
Here's `MoveItemToTrash`'s implementation https://github.com/electron/electron/blob/f7e3f9035da3bf3eb946294a4100b6700bd41bc4/atom/common/platform_util_linux.cc#L98 which calls into this https://github.com/electron/electron/blob/eb63bea87c299b3c440d6a36ccced3cc9653e1c4/atom/common/platform_util_linux.cc#L20

The logic seems to go:

1. Use `kioclient5` or `kioclient` if available
2. Fallback to `trash-cli` if it's available (this? https://github.com/andreafrancia/trash-cli)
3. Fallback to `gvfs-trash`

@kbumsik could you check if `gvfs-trash` is installed, and if so try use it to perform the same trash action that is failing? I believe you just want to do the call like this:

```
gvfs-trash /path/to/failing/file
```

**Comment by kbumsik** [comment]
@bpasero Shift+Delete works alternatively, thanks :)
@Tyriar I am running VS Code on a local machine and I tried this under the home directory. The permission shouldn't be the case. Next, I installed `gvfs-bin`, which is a package associated with `gvfs-trash` and tried deleting the file on VS code. It works well without any problem!
Looks like the problem is `gvfs` is not installed in Kubuntu because Kubuntu is KDE based distribution.
